stock_landed_costs_delivery/models/stock_picking.py

```diff
@@ -7,6 +7,12 @@
 class StockPicking(models.Model):
     _inherit = "stock.picking"
 
+    def _add_delivery_cost_to_po(self):
+        """We add the context to identify in _create_delivery_line() method the picking."""
+        self.ensure_one()
+        self = self.with_context(from_picking_id=self)
```

Comment:
If you pass a recordset, be consistent with the context key name:

```suggestion
        self = self.with_context(from_picking=self)
```